### PR TITLE
Restyle top and guide section

### PR DESCRIPTION
**Pull request by ludeeus**
## Description:

An attempt to make the top part of the landing page a bit more "friendly"? "modern"?

- Moves the discord/github/twitter badges to the sidebar as links (which now open a new tab)
- Changes the guide section from the `imgtable` directive to a card format
- Adds a hero (overview) image

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Review by jesserockz — APPROVED** [review]
I like it.

Maybe we could do something about there being 3 ESPHome's at the top?

Screenshot from 2021-06-15 20-24-39

**Comment by ludeeus** [comment]
Removed the static header, the big logo make sense on the mainpage (at least with the current layout), and the logo above the navigation makes sense on all other pages, but removing it from the main page only makes it non-consistent

**Comment by frenck** [comment]
Nice!

image

Maybe remove "With" from the add-on, as it linebreaks

**Comment by balloob** [comment]
Can we move Github/Discord/Twitter below 	the new cards

Can we add below the introduction a simple thing you can do with ESPHome. It's still not clear. Maybe something that boils down to this (but a bit more polished):

image


(taken from https://deploy-preview-1242--esphome.netlify.app/components/sensor/dht.html)

**Comment by balloob** [comment]
This looks great ! Can we make the configuration bigger? It's a bit hard to read now. Same with the HA Card, maybe we can make both less wide. Config has unused space on the right, card has unused space in the middle.

File: index.rst [hunk 1]
```diff
@@ -14,32 +11,83 @@ ESPHome
...
+                <li>
+                    <a class="reference" href="/guides/getting_started_command_line.html">
+                        Through Command Line
+                    </a>
+                </li>
+                <li>
+                    <a class="reference" href="/guides/getting_started_hassio.html">
+                        Home Assistant Add-on
+                    </a>
+                </li>
```
Comment (balloob) [inline index.rst]:
```suggestion
                <li>
                    <a class="reference" href="/guides/getting_started_hassio.html">
                        ESPHome Dashboard
                    </a>
                </li>
                <li>
                    <a class="reference" href="/guides/getting_started_command_line.html">
                        Command Line Interface
                    </a>
                </li>
```

**Comment by glmnet** [comment]
> I like it.
> 
> 
> 
> Maybe we could do something about there being 3 ESPHome's at the top?
> 
> 
> 
> Screenshot from 2021-06-15 20-24-39
> 
> 

Jaja tried this before and failed miserably

**Comment by balloob** [comment]
It’s down to two now. Improvement! :P we can give the last logo a shot in a new PR .

**Comment by ludeeus** [comment]
https://github.com/esphome/esphome-docs/pull/1242#issuecomment-861299398 are my reasons to keep the remaining

**Pull request merged**